assets/src/state/searchPageState.ts

```diff
@@ -143,17 +153,23 @@ export const reducer = (
   action: Action
 ): SearchPageState => {
   switch (action.type) {
-    case "SET_SEARCH_TEXT":
+    case "SET_SEARCH_TEXT": {
       return {
         ...state,
         query: {
           ...state.query,
           text: action.payload.text,
-          properties: defaultAllProperties,
+          properties: Object.fromEntries(
```

Comment:
This logic and the logic in the `setSearchPropertyLimits` limits has me wondering if it might be worth changing the representation in state to a list `{property: property, limit: limit}[]`, where the only entries in the list are ones where the filters are toggled on, and toggling off a filter would result in removing an item from the list. With the current representation, is limit doing too much by representing both the toggled state (limit === 0) and whether the search result limit has been increased?

Reply:
Hmm, my one concern with a list representation is that it doesn't automatically enforce certain things (like having at most entry with a given `property`).

One random thought: how about instead of `limit = 0` to represent an un-toggled property we use `null` for that purpose? That makes it a bit clearer that we're not dealing with a normal property limit.

Reply:
Good points - I'll switch to `null`!